format/wasm/mappers.go

```diff
@@ -0,0 +1,211 @@
+package wasm
+
+import (
+	"errors"
+
+	"github.com/wader/fq/pkg/scalar"
+)
+
+var sectionIDToSym = &sectionIDToSymMapper{}
+
+type sectionIDToSymMapper struct {
+}
+
+func (m *sectionIDToSymMapper) MapScalar(s scalar.S) (scalar.S, error) {
+	v, ok := s.Actual.(uint64)
+	if !ok {
+		return s, errors.New("unexpected data type for section ID")
+	}
+
+	switch v {
+	case sectionIDCustom:
+		s.Sym = "custom section"
+	case sectionIDType:
+		s.Sym = "type section"
+	case sectionIDImport:
+		s.Sym = "import section"
+	case sectionIDFunction:
+		s.Sym = "function section"
+	case sectionIDTable:
+		s.Sym = "table section"
+	case sectionIDMemory:
+		s.Sym = "memory section"
+	case sectionIDGlobal:
+		s.Sym = "global section"
+	case sectionIDExport:
+		s.Sym = "export section"
+	case sectionIDStart:
+		s.Sym = "start section"
+	case sectionIDElement:
+		s.Sym = "element section"
+	case sectionIDCode:
+		s.Sym = "code section"
+	case sectionIDData:
+		s.Sym = "data section"
+	case sectionIDDataCount:
+		s.Sym = "data count section"
+	default:
+		s.Sym = "unknown section"
+	}
+
+	return s, nil
+}
+
+var valtypeToSymMapper = &valtypeToSym{}
+
+type valtypeToSym struct {
+}
+
+// valtype ::= t:numtype => t
+//          |  t:vectype => t
+//          |  t:reftype => t
+func (m *valtypeToSym) MapScalar(s scalar.S) (scalar.S, error) {
+	v, ok := s.Actual.(uint64)
+	if !ok {
+		return s, errors.New("unexpected data type for valtype")
+	}
+
+	switch v {
+	case 0x7f:
+		s.Sym = "i32"
+	case 0x7e:
+		s.Sym = "i64"
+	case 0x7d:
+		s.Sym = "f32"
+	case 0x7c:
+		s.Sym = "f64"
+	case 0x7b:
+		s.Sym = "v128"
+	case 0x70:
+		s.Sym = "funcref"
+	case 0x6f:
+		s.Sym = "externref"
+	default:
+		s.Sym = "unknown valtype"
```

Comment:
Some of these mappers might be possible to express using the predefined table mappers
```go
var valtypeMapper = scalar.UToSymStr{
	0x7f:   "i32",
        // ...
}
```

Reply:
Maybe could be nice to add constant for the number literals? but it depends a bit i guess, i usually add constant if the number(s) is used in severals places or if i want to do something more fancy with them

Reply:
modified to use `scalar.UToSymStr` @ https://github.com/wader/fq/pull/383/commits/63f4a7267c3f4920bea98baea736663831afa417

It's way simple!

Reply:
Yeah they are nice, there are from/to table mappers for most types supported